doc/specs/wiki.md

```diff
@@ -0,0 +1,222 @@
+2. The user makes the appropriate changes to the title, body, and user class read/edit restrictions.
+3. The user clicks submit.
+4. The changes are saved to the database, and the user is redirected to the updated article.
+
+**Functional Requirements:**
+
+* The article's revision history must be updated with the changes.
+
+### 3.1 Previewing articles
+
+**Primary Actor:** A user
+
+**Brief:** When editing an article, previewing ensures that the formatting is correct.
+
+**Preconditions:** The user is creating an article, or is on the edit page of an article.
+
+**Trigger:** The user clicks the preview button.
+
+**Postcondition:**
+
+* The article is rendered in-place, without committing anything to the database.
+
+**Basic Flow:**
+
+1. The user is making changes to an article, or writing a new one.
+2. The preview button is clicked.
+3. The article is rendered in-place, and is presented to the user.
+
+**Functional Requirements:**
+
+* BBCode must be rendered when the article is previewed.
+
+### 4.0 Adding aliases
+
+**Primary Actor:** A user
+
+**Brief:** Alises allow wiki articles to be easily searched. Managing aliases ensures they are accurate and thorough.
+
+**Preconditions:** The user is currently viewing the article.
+
+**Trigger:** The user clicks the submit button next to the alias textbox.
+
+**Postcondition:**
+
+* The alias is added to the article.
+
+**Basic Flow:**
+
+1. The user enters the new alias into the textbox.
+2. The user hits submit, and the alias is added to the article.
+
+**Functional Requirements:**
+
+* Duplicate and null aliases must be able to be detected, and discarded.
+* New aliases are committed to the database.
+
+### 4.1 Removing aliases
+
+**Primary Actor:** A user
+
+**Brief:** Aliases must be able to ensure relevancy and accuracy.
+
+**Preconditions:** The user is currently viewing the article.
+
+**Trigger:** The user clicks the X button next to an alias.
+
+**Postcondition:**
+
+* The alias is removed.
+
+**Basic Flow:**
+
+1. The user clicks the X button next to an alias.
+2. The alias is removed.
+
+**Functional Requirements:**
+
+* The alias must be removed from the database.
+
+### 4.2 Viewing a user that added an alias
+
+**Primary Actor:** A user
+
+**Brief:** Users must be able to view which user added an alias.
+
+**Preconditions:** N/A
+
+**Trigger:** The user clicks the U button next to an alias.
+
+**Postconditions:** The user is redirected to the user that added the alias.
+
+**Basic Flow:**
+
+1. The user clicks the U button next to an alias.
+2. The user is redirected to the user that added the alias.
+
+## Data model
+
+What data should be tracked.
+
+**Wiki Articles:**
+> * The title of the article.
+> * The body of the article, containing the BBCode tags.
+> * The current revision number.
+> * The ID of the user that created the article.
+> * The timestamp of the article creation time.
+> * The timestamp of the last edit time.
+
+**Wiki Revisions:**
```

Comment:
Also this would be stored in the wiki document. maybe we could store a series of diffs for the revisions?